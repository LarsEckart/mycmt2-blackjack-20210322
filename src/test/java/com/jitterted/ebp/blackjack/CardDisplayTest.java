package com.jitterted.ebp.blackjack;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class CardDisplayTest {

  @Test
  void displayTenAsString() {
    Card card = new Card(Suit.HEARTS, Rank.TEN);

    Approvals.verify(card.display());
  }

  @Test
  void displayNonTenAsString() {
    Card card = new Card(Suit.CLUBS, Rank.SEVEN);

    Approvals.verify(card.display());
  }

  @Test
  void all_combinations() {
    CombinationApprovals.verifyAllCombinations(this::displayCard, Suit.values(), Rank.values());
  }

  private String displayCard(Suit suit, Rank rank) {
    return new Card(suit, rank).display();
  }
}
