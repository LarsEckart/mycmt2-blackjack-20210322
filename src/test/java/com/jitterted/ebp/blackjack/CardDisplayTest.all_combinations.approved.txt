[HEARTS, ACE] => [31m┌─────────┐[1B[11D│A        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        A│[1B[11D└─────────┘ 
[HEARTS, TWO] => [31m┌─────────┐[1B[11D│2        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        2│[1B[11D└─────────┘ 
[HEARTS, THREE] => [31m┌─────────┐[1B[11D│3        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        3│[1B[11D└─────────┘ 
[HEARTS, FOUR] => [31m┌─────────┐[1B[11D│4        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        4│[1B[11D└─────────┘ 
[HEARTS, FIVE] => [31m┌─────────┐[1B[11D│5        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        5│[1B[11D└─────────┘ 
[HEARTS, SIX] => [31m┌─────────┐[1B[11D│6        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        6│[1B[11D└─────────┘ 
[HEARTS, SEVEN] => [31m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘ 
[HEARTS, EIGHT] => [31m┌─────────┐[1B[11D│8        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        8│[1B[11D└─────────┘ 
[HEARTS, NINE] => [31m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘ 
[HEARTS, TEN] => [31m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘ 
[HEARTS, JACK] => [31m┌─────────┐[1B[11D│J        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        J│[1B[11D└─────────┘ 
[HEARTS, QUEEN] => [31m┌─────────┐[1B[11D│Q        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        Q│[1B[11D└─────────┘ 
[HEARTS, KING] => [31m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♥    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘ 
[CLUBS, ACE] => [30m┌─────────┐[1B[11D│A        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        A│[1B[11D└─────────┘ 
[CLUBS, TWO] => [30m┌─────────┐[1B[11D│2        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        2│[1B[11D└─────────┘ 
[CLUBS, THREE] => [30m┌─────────┐[1B[11D│3        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        3│[1B[11D└─────────┘ 
[CLUBS, FOUR] => [30m┌─────────┐[1B[11D│4        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        4│[1B[11D└─────────┘ 
[CLUBS, FIVE] => [30m┌─────────┐[1B[11D│5        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        5│[1B[11D└─────────┘ 
[CLUBS, SIX] => [30m┌─────────┐[1B[11D│6        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        6│[1B[11D└─────────┘ 
[CLUBS, SEVEN] => [30m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘ 
[CLUBS, EIGHT] => [30m┌─────────┐[1B[11D│8        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        8│[1B[11D└─────────┘ 
[CLUBS, NINE] => [30m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘ 
[CLUBS, TEN] => [30m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘ 
[CLUBS, JACK] => [30m┌─────────┐[1B[11D│J        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        J│[1B[11D└─────────┘ 
[CLUBS, QUEEN] => [30m┌─────────┐[1B[11D│Q        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        Q│[1B[11D└─────────┘ 
[CLUBS, KING] => [30m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♣    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘ 
[DIAMONDS, ACE] => [31m┌─────────┐[1B[11D│A        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        A│[1B[11D└─────────┘ 
[DIAMONDS, TWO] => [31m┌─────────┐[1B[11D│2        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        2│[1B[11D└─────────┘ 
[DIAMONDS, THREE] => [31m┌─────────┐[1B[11D│3        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        3│[1B[11D└─────────┘ 
[DIAMONDS, FOUR] => [31m┌─────────┐[1B[11D│4        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        4│[1B[11D└─────────┘ 
[DIAMONDS, FIVE] => [31m┌─────────┐[1B[11D│5        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        5│[1B[11D└─────────┘ 
[DIAMONDS, SIX] => [31m┌─────────┐[1B[11D│6        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        6│[1B[11D└─────────┘ 
[DIAMONDS, SEVEN] => [31m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘ 
[DIAMONDS, EIGHT] => [31m┌─────────┐[1B[11D│8        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        8│[1B[11D└─────────┘ 
[DIAMONDS, NINE] => [31m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘ 
[DIAMONDS, TEN] => [31m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘ 
[DIAMONDS, JACK] => [31m┌─────────┐[1B[11D│J        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        J│[1B[11D└─────────┘ 
[DIAMONDS, QUEEN] => [31m┌─────────┐[1B[11D│Q        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        Q│[1B[11D└─────────┘ 
[DIAMONDS, KING] => [31m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♦    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘ 
[SPADES, ACE] => [30m┌─────────┐[1B[11D│A        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        A│[1B[11D└─────────┘ 
[SPADES, TWO] => [30m┌─────────┐[1B[11D│2        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        2│[1B[11D└─────────┘ 
[SPADES, THREE] => [30m┌─────────┐[1B[11D│3        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        3│[1B[11D└─────────┘ 
[SPADES, FOUR] => [30m┌─────────┐[1B[11D│4        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        4│[1B[11D└─────────┘ 
[SPADES, FIVE] => [30m┌─────────┐[1B[11D│5        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        5│[1B[11D└─────────┘ 
[SPADES, SIX] => [30m┌─────────┐[1B[11D│6        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        6│[1B[11D└─────────┘ 
[SPADES, SEVEN] => [30m┌─────────┐[1B[11D│7        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        7│[1B[11D└─────────┘ 
[SPADES, EIGHT] => [30m┌─────────┐[1B[11D│8        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        8│[1B[11D└─────────┘ 
[SPADES, NINE] => [30m┌─────────┐[1B[11D│9        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        9│[1B[11D└─────────┘ 
[SPADES, TEN] => [30m┌─────────┐[1B[11D│10       │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│       10│[1B[11D└─────────┘ 
[SPADES, JACK] => [30m┌─────────┐[1B[11D│J        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        J│[1B[11D└─────────┘ 
[SPADES, QUEEN] => [30m┌─────────┐[1B[11D│Q        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        Q│[1B[11D└─────────┘ 
[SPADES, KING] => [30m┌─────────┐[1B[11D│K        │[1B[11D│         │[1B[11D│    ♠    │[1B[11D│         │[1B[11D│        K│[1B[11D└─────────┘ 
